### ISSUE TITLE
[BUG] doom/increase-font-size increases modeline

### ISSUE THREAD

**Issue by ijacquez** [hlissner/doom-emacs#2186]
**What did you expect to happen?**
Increase the font size everywhere via `M-=`, but keep the modeline size _proportional_ to the font size. This might be related to [#575](https://github.com/hlissner/doom-emacs/issues/575).

**What actually happened?**
Increasing the font size works 100%, however the modeline is too tall. 

**Additional details:**
See an extreme example:
![image](https://user-images.githubusercontent.com/283713/70740314-bd6afc00-1ccd-11ea-99cc-fdf5e79bf9b4.png)

**Steps to reproduce:**
1. Start Emacs
2. Keep pressing `M-=`

**System information:**
<details>

```
emacs   version    26.1
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 THREADS LIBSYSTEMD LCMS2
        build      Apr 11, 2019
        buildopts  (--build x86_64-linux-gnu --prefix=/usr --sharedstatedir=/var/lib --libexecdir=/usr/lib --localstatedir=/var/lib --infodir=/usr/share/info --mandir=/usr/share/man --enable-libsystemd --with-pop=yes --enable-locallisppath=/etc/emacs:/usr/local/share/emacs/26.1/site-lisp:/usr/local/share/emacs/site-lisp:/usr/share/emacs/26.1/site-lisp:/usr/share/emacs/site-lisp --with-sound=alsa --without-gconf --with-mailutils --build x86_64-linux-gnu --prefix=/usr --sharedstatedir=/var/lib --libexecdir=/usr/lib --localstatedir=/var/lib --infodir=/usr/share/info --mandir=/usr/share/man --enable-libsystemd --with-pop=yes --enable-locallisppath=/etc/emacs:/usr/local/share/emacs/26.1/site-lisp:/usr/local/share/emacs/site-lisp:/usr/share/emacs/26.1/site-lisp:/usr/share/emacs/site-lisp --with-sound=alsa --without-gconf --with-mailutils --with-x=yes --with-x-toolkit=gtk3 --with-toolkit-scroll-bars 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/emacs-cQxEw2/emacs-26.1+1=. -fstack-protector-strong -Wformat -Werror=format-security -Wall' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro')
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD 318754c4d 2019-12-05 20:48:53 -0500
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/fish
        uname      Linux 5.0.0-31-generic #33-Ubuntu SMP Mon Sep 30 18:51:59 UTC 2019 x86_64
        path       (~/.local/bin /snap/bin /usr/lib/cargo/bin ~/x-tools/i686-w64-mingw32/bin ~/x-tools/i686-pc-linux-gnu/bin ~/x-tools/x86_64-w64-mingw32/bin ~/x-tools/sh2eb-elf/bin ~/.local/bin /snap/bin /usr/lib/cargo/bin ~/x-tools/i686-w64-mingw32/bin ~/x-tools/i686-pc-linux-gnu/bin ~/x-tools/x86_64-w64-mingw32/bin ~/x-tools/sh2eb-elf/bin ~/.local/bin /snap/bin /usr/lib/cargo/bin ~/x-tools/i686-w64-mingw32/bin ~/x-tools/i686-pc-linux-gnu/bin ~/x-tools/x86_64-w64-mingw32/bin ~/x-tools/sh2eb-elf/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /usr/lib/emacs/26.1/x86_64-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion (company +auto) ivy :ui workspaces doom doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) unicode vc-gutter vi-tilde-fringe window-select :editor snippets fold multiple-cursors rotate-text :emacs dired electric vc :tools (lookup +docsets) eval editorconfig flycheck flyspell magit rgb :lang assembly cc common-lisp csharp data emacs-lisp markdown (org +dragndrop +attach +babel +capture +export +present) python (sh +fish) :config (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>

**Comment by hlissner** [comment]
What have you set `doom-font` to?

**Comment by ijacquez** [comment]
I'm setting no font explicitly.

One more piece of info that I forgot to mention. I'm running Emacs (UI) via SSH forwarding on Windows.

**Comment by hlissner** [comment]
Does the issue persist if you do explicitly set a font?

**Comment by ijacquez** [comment]
I forgot to mention that I have `(setq doom-font-increment 4)` set.

I removed it, and it looks less extreme.

I set `(setq doom-font (font-spec :family "Fira Mono" :size 19))` with `doom-font-increment`, and it doesn't have the same problem.

It seems like `doom-font-increment` is what causes the issue.

Below is setting `doom-font-increment` to `16` and pressing `M-=` once, with `doom-font` set.
![image](https://user-images.githubusercontent.com/283713/70868283-19cf5500-1f34-11ea-974e-9e24a6599fe1.png)
